framework/src/controller/validator/keywords/env/index.js

```diff
@@ -18,7 +18,23 @@ const compile = (schema, parentSchema) => {
 			  };
 
 	return function(data, dataPath, object, key) {
-		const variableValue = process.env[envVariable.name];
+		let variableValue = process.env[envVariable.name];
+
+		// Formatting logic to automatically format the CLI value to the expected type
+		// Only format if the variable exist
+		if (variableValue !== undefined) {
+			// eslint-disable-next-line default-case
+			switch (parentSchema.type) {
+				case 'array':
+					variableValue = variableValue.split(',');
```

Comment:
This change will break the following implementation. 

https://github.com/LiskHQ/lisk-sdk/blob/32950c965f96e8b54bcaf9a6667089a11318dc7b/framework/src/modules/network/defaults/config.js#L49

https://github.com/LiskHQ/lisk-sdk/blob/32950c965f96e8b54bcaf9a6667089a11318dc7b/framework/src/controller/validator/keywords/formatters.js#L18-L28

I suggest not to parse array values on this level. Leave it to actual implementation. String to array conversion can be different in different cases.